### DISCUSSION TITLE
Slow startServer on TestintApp

### DISCUSSION THREAD

**Discussion by rjurado01** [marcj/deepkit#422] [Q&A] [answered]
I have created a new deepkit app and have edited `tests/http.spec.ts` to show how long does it take to start the app:

```ts
test('http controller', async () => {
    const testing = createTestingApp({
        controllers: [HelloWorldControllerHttp],
        providers: [Service]
    });

    console.time()
    await testing.startServer();
    console.timeEnd()

    try {
        const response = await testing.request(HttpRequest.GET('/hello/World'));
        expect(response.statusCode).toBe(200);
        expect(response.json).toBe("Hello World!");
    } finally {
        await testing.stopServer();
    }
});
```

It show me: `685 ms`

If I do the same using NestJs (creates new app and measure the time):

```ts
describe('AppController (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();

    console.time();
    await app.init();
    console.timeEnd();
  });

  it('/ (GET)', () => {
    return request(app.getHttpServer())
      .get('/')
      .expect(200)
      .expect('Hello World!');
  });
});
```

It shows me `5ms`

Why is so slow deepkit?

**Comment by marcj** [comment, marked as answer]
Is it every time or only the very first one? You can try it in a for loop to verify

**Reply by rjurado01** [reply]
It is true, second and other times it is faster:

![image](https://user-images.githubusercontent.com/1703608/224417729-3db97481-8643-4b61-a1d3-e8228d2c39ed.png)

but Nestjs is even faster

![image](https://user-images.githubusercontent.com/1703608/224417893-dcc5a3be-4940-45aa-b89b-b5f75087e334.png)

**Reply by marcj** [reply]
That's fine. We don't optimise for the server start as it's pointless.
